File: Clase6/SoftProgSolution/SoftProgNegocio/src/main/java/pe/edu/pucp/softprog/bo/IBaseBO.java
package pe.edu.pucp.softprog.bo;

import java.util.List;

public interface IBaseBO <T>{
    int insertar(T objeto) throws Exception;
    int modificar(T objeto) throws Exception;
    int eliminar(int id) throws Exception;
    List<T> listarTodos() throws Exception;
    T buscarPorId(int id) throws Exception;
}

File: Clase12/ProyectosNET/SoftProgSolution/SoftProgNegocio/BO/IBaseBO.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace SoftProgNegocio.BO
{
    public interface IBaseBO<T>
    {
        int insertar(T objeto);
        int modificar(T objeto);
        int eliminar(int idObjeto);
        List<T> listarTodos();
        T buscarPorId(int id);
    }
}
